dio/lib/src/sentry_transformer.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+/// A transformer which wraps transforming in spans
+class SentryTransformer implements Transformer {
+  // ignore: public_member_api_docs
+  SentryTransformer({required Transformer transformer, Hub? hub})
+      : _hub = hub ?? HubAdapter(),
+        _transformer = transformer;
+
+  final Transformer _transformer;
+  final Hub _hub;
+
+  @override
+  Future<String> transformRequest(RequestOptions options) async {
+    final span = _hub.getSpan()?.startChild(
+          'serialize',
+          description: 'Dio.transformRequest: ${options.method} ${options.uri}',
+        );
+    try {
+      final request = await _transformer.transformRequest(options);
+      await span?.finish(status: SpanStatus.ok());
+      return request;
+    } catch (_) {
+      await span?.finish(status: SpanStatus.internalError());
+      rethrow;
```

Comment:
I'd rather use the convention used by `TracingClient`, which is the try/catch/finally to set status, exception, rethrow and finish